packages/pwa-kit-dev/src/ssr/server/loading-screen/index.html

```diff
@@ -86,8 +86,9 @@ <h1 class="loading-text">Building your app</h1>
                 }
 
                 waitForReady().then(() => {
-                    // Redirect to homepage
-                    window.location = url.origin
+                    // Redirect to homepage if path is not defined
+                    const path = url.searchParams.get('path') ?? ''
+                    window.location = url.origin + path
```

Comment:
🙌 loading screen not redirecting to the original page has been such a minor nuisance for so long!